src/App.js

```diff
@@ -0,0 +1,111 @@
+import DocumentList from "./component/DocumentList.js";
+import Document from "./component/Document.js";
+import { initRouter, push } from "./core/router.js";
+import nestedObjectToArray from "./core/nestedObjectToArray.js";
+import { getItem, setItem } from "./core/storage.js";
+import { request } from "./core/api.js";
+import editor from "./component/Editor.js";
+
+export default function App({ $target }) {
+  this.state = getItem("tempSaveKey") || alert("새로고침을 해주세요");
+
+  this.setState = async (nextState) => {
+    this.state = nextState;
+    await documentList.setState(this.state);
+    await document.setState(this.state);
+  };
+
+  this.fetchDocument = async () => {
+    if (getItem("tempSaveKey") == null) {
+      const serverDocument = nestedObjectToArray(
+        await request("/documents", {
+          method: "GET",
+        })
+      );
+      setItem("tempSaveKey", {
+        documentList: serverDocument,
+        currentDocumentId: serverDocument[0].id,
+      });
+    }
+    await this.setState(getItem("tempSaveKey"));
+  };
+
+  this.fetchDocument();
+
+  const documentList = new DocumentList({
+    $target,
+    initialState: this.state,
+    onClick: async (clickedId) => {
+      const nextState = {
+        documentList: nestedObjectToArray(
+          await documentList.fetchDocumentList()
+        ),
+        currentDocumentId: clickedId,
+      };
+      setItem("tempSaveKey", nextState);
+    },
+    onPost: async (clickedId) => {
+      await request("/documents", {
+        method: "POST",
+        body: JSON.stringify({
+          title: "문서 제목",
+          parent: clickedId,
+        }),
+      });
+      const nextState = {
+        documentList: nestedObjectToArray(
+          await documentList.fetchDocumentList()
+        ),
+        currentDocumentId:
+          clickedId == "null"
+            ? getItem("tempSaveKey").currentDocumentId
+            : clickedId,
+      };
+      this.setState(nextState);
+      setItem("tempSaveKey", nextState);
+    },
+    onDelete: async (clickedId) => {
+      await request(`/documents/${clickedId}`, {
+        method: "DELETE",
+      });
+      const nextState = {
+        documentList: nestedObjectToArray(
+          await documentList.fetchDocumentList()
+        ),
+        currentDocumentId:
+          getItem("tempSaveKey").documentList[
+            getItem("tempSaveKey").documentList.findIndex(
+              (document) => document.id == clickedId
```

Comment:
```suggestion
              (document) => document.id === Number(clickedId)
```
`==`으로 하면 동일한 값이지만 다른 타입인 값이 같은 값으로 인식이 될 수 있기 때문에 위와 같이 형변환을 해주고 "엄격한" 비교를 해주시면 좋을 것 같습니다. [동치 비교 및 동일성](https://developer.mozilla.org/ko/docs/Web/JavaScript/Equality_comparisons_and_sameness) 참고하시면 좋을 것 같아요!! 😀